object_test.go

```diff
@@ -1006,3 +1018,267 @@ func TestObjectFilter(t *testing.T) {
 			object.chain.assertNotFailed(t)
 		})
 }
+
+func TestObjectFind(t *testing.T) {
+	t.Run("Find an object of elements of the same type and validate", func(ts *testing.T) {
+		reporter := newMockReporter(t)
+		object := NewObject(reporter, map[string]interface{}{
+			"foo":  "bar",
+			"baz":  "qux",
+			"quux": "corge",
+		})
+		foundValue := object.Find(func(key string, value *Value) bool {
+			return key == "baz"
+		})
+		assert.Equal(t, "qux", foundValue.Raw())
+		assert.Equal(t, object.Raw(), map[string]interface{}{
+			"foo":  "bar",
+			"baz":  "qux",
+			"quux": "corge",
+		})
+
+		foundValue.chain.assertNotFailed(t)
+		object.chain.assertNotFailed(t)
+	})
+
+	t.Run("Find an object of elements of the multi type and validate", func(ts *testing.T) {
+		reporter := newMockReporter(t)
+		object := NewObject(reporter, map[string]interface{}{
+			"foo":  "bar",
+			"baz":  true,
+			"qux":  -1,
+			"quux": 2,
+		})
+		foundValue := object.Find(func(key string, value *Value) bool {
+			n := value.Number().Raw()
+			return n > 1
+		})
+		assert.Equal(t, 2.0, foundValue.Raw())
+		assert.Equal(t, object.Raw(), map[string]interface{}{
+			"foo":  "bar",
+			"baz":  true,
+			"qux":  -1.0,
+			"quux": 2.0,
+		})
+
+		foundValue.chain.assertNotFailed(t)
+		object.chain.assertNotFailed(t)
+	})
+
+	t.Run("No match", func(ts *testing.T) {
+		reporter := newMockReporter(t)
+		object := NewObject(reporter, map[string]interface{}{
+			"foo":  "bar",
+			"baz":  true,
+			"qux":  -1,
+			"quux": 2,
+		})
+		foundValue := object.Find(func(key string, value *Value) bool {
+			n := value.Number().Raw()
+			return n == 3
+		})
+		assert.Equal(t, []interface{}{}, foundValue.Raw())
+		assert.Equal(t, object.Raw(), map[string]interface{}{
+			"foo":  "bar",
+			"baz":  true,
+			"qux":  -1.0,
+			"quux": 2.0,
+		})
+
+		foundValue.chain.assertFailed(t)
+		object.chain.assertFailed(t)
+	})
+
+	t.Run("Empty Object", func(ts *testing.T) {
+		reporter := newMockReporter(t)
+		object := NewObject(reporter, map[string]interface{}{})
+		foundValue := object.Find(func(key string, value *Value) bool {
+			n := value.Number().Raw()
+			return n == 3
+		})
+		assert.Equal(t, []interface{}{}, foundValue.Raw())
+		assert.Equal(t, object.Raw(), map[string]interface{}{})
+
+		foundValue.chain.assertFailed(t)
+		object.chain.assertFailed(t)
+	})
+}
+func TestObjectFindAll(t *testing.T) {
+	t.Run("Find values in array of the same type", func(ts *testing.T) {
+		reporter := newMockReporter(t)
+		object := NewObject(reporter, map[string]interface{}{
+			"foo":  "bar",
+			"baz":  "qux",
+			"quux": "corge",
+		})
+		foundValues := object.FindAll(func(key string, value *Value) bool {
+			return key == "baz" || key == "quux"
+		})
+
+		actual := []interface{}{}
+		for _, value := range foundValues {
+			actual = append(actual, value.Raw())
+		}
+
+		assert.ElementsMatch(t, []interface{}{"qux", "corge"}, actual)
```

Comment:
`map[string]interface{}` has randomize iteration order, so I use `assert.ElementsMatch`

Reply:
Thanks for this notice!

In this lib I try to make all methods deterministic. Object.Keys and Object.Values return elements sorted by key. Let's use the same approach here: sort values by key before looking for them, in all three methods. This way results and error messages will be the same each run.

Reply:
I've also created related issue https://github.com/gavv/httpexpect/issues/212

Reply:
fixed [48d4b10](https://github.com/gavv/httpexpect/pull/211/commits/48d4b1018a8acfff08613fb8f35198489df2719f) 
sort values by key before looking for them.